src/App.tsx

```diff
@@ -1,34 +1,72 @@
-/* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
-
+import React, { useEffect, useState } from 'react';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions } from './features/todos';
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 
 export const App: React.FC = () => {
+  const [error, setError] = useState<string>('');
+  const openedTodo = useAppSelector(state => state.currentTodo);
+  const todoList = useAppSelector(state => state.todos);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const loadData = async () => {
+      try {
+        const todos = await getTodos();
+        const setTodos = () => dispatch(actions.setTodos(todos));
+
+        setTodos();
+      } catch {
+        setError('We can not load todos.');
+      }
+    };
+
+    loadData();
+  }, []);
+
   return (
     <>
       <div className="section">
         <div className="container">
           <div className="box">
-            <h1 className="title">Todos:</h1>
+            {error
+              ? (
+                <p className="notification is-warning">
+                  Something went wrong!
+                </p>
+              )
+              : (
+                <>
+                  <h1 className="title">Todos:</h1>
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+                  <div className="block">
+                    <TodoFilter />
+                  </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+                  <div className="block">
+                    {
+                      todoList.length > 0
+                        ? (
+                          <TodoList />
+                        )
+                        : <Loader />
+                    }
```

Comment:
let's change it a little bit for better readability
```suggestion
                    {todoList.length > 0
                      ? <TodoList />
                      : <Loader />}
```